subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/FlakinessIndexPageGenerator.java

```diff
@@ -114,8 +111,9 @@ protected void renderScenarioButtons(int index, ScenarioBuildResultData scenario
     }
 
     public void reportToIssueTracker() {
-        scenarios.stream().filter(FlakinessIndexPageGenerator::isFlaky).forEach(issueReporter::report);
+        scenarios.stream().filter(FlakinessIndexPageGenerator::isFlaky).forEach(PerformanceFlakinessAnalyzer.getInstance()::report);
```

Comment:
Why did we switch from service injection to the singleton pattern?

Reply:
The intention is to avoid creating more than one instances by accident, since reading flaky issues from GH is a very expensive operation.

Reply:
Can we still use a service? We only ever run this code once per build, right?

Reply:
Previously we only use the service in one place, now we're using it in two places.

Reply:
Yes. Though both of the places run in a separate VM since they are started via `JavaExec`, right? So a static field does not help much.

Reply:
Ah, I mean if we keep using service, we need to create the service instance in `AbstractReportGenerator`, then pass it via constructor chain:

1. `IndexPageGenerator` -> `AbstractTablePageGenerator`,
2. `FlakinessReportGenerator` -> `FlakinessIndexPageGenerator` -> `AbstractTablePageGenerator`

We can sure keep it as a service, but I think this chain is not convenient enough. I'd like to use singleton here. WDYT?

Reply:
I wonder how we cache the issues at all. `GitHubKnownIssuesProvider.knownInvalidFailures` does the query each time it is called, right? So the creation is not expensive, only the querying.

Reply:
I made the change we avoids the singleton pattern and another one which caches the known flaky tests like we do in `InvalidFailureErrorAnalyzer`. Please take a look.